saleor/order/utils.py

```diff
@@ -244,30 +237,20 @@ def update_taxes_for_order_lines(
     )
 
 
-def update_order_prices(
+def update_order_prices_if_expired(
```

Comment:
Why do we call `update_order_lines_prices_if_expired`  and `fetch_order_prices_if_expired`? I thought that new way of fetching prices will always calculate all prices from a beginning (if expired). Can't we just call `fetch_order_prices_if_expired` once here? 
Also, maybe I am missing something but I am not convinced that we should mix in a calculation of order prices with invalidating them. 
I would say that current flow is incorrect, especially with flag invalidate_prices=True:
1. call update_order_lines_prices_if_expired
2. call fetch_order_prices_if_expired (which I believe calls the same logic as in 1st point)
3. call `recalculate_order` with invalidate_prices=True which means that this will overwrite and force to re-calculate all prices once again. 

Also, I am not sure that invalidating prices is something that belongs to `recalculate_order`.

Reply:
I don't think that we still need functions like this one. There is no need to fetch prices in advance.  If we need to have a prices we can use a interface to just take them like `order.calculation.unit_price`. 
Mutations for draf/unconfirmed orders should invalidate a price expiration but nothing more.